subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -126,40 +141,81 @@ The code for resolving the dependencies should be moved into the task action to
 
 ++++
 <sample id="conditionalLogicDont" dir="userguide/bestPractices/conditionalLogic/dont" title="A build script using conditional logic to create a task">
     <sourcefile file="build.gradle"/>
 </sample>
 ++++
 
-Let's compare the build script with the same logic implemented as a binary plugin. The code might look more involved at first but clearly looks more like typical application code. This particular plugin class lives in the <<sec:build_sources,`buildSrc` directory>> which makes it available to the build script automatically.
+Let's compare the build script with the same logic implemented as a binary plugin.
+The code might look more involved at first but clearly looks more like typical application code.
+This particular plugin class lives in the <<sec:build_sources,`buildSrc` directory>> which makes it available to the build script automatically.
 
 ++++
 <sample id="conditionalLogicDo" dir="userguide/bestPractices/conditionalLogic/do/buildSrc/src/main/java/com/enterprise" title="A binary plugin implementing imperative logic">
     <sourcefile file="ReleasePlugin.java"/>
 </sample>
 ++++
 
-Now that the build logic has been translated into a plugin, you can apply it in the build script. The build script has been shrunk from 8 lines of code to a one liner.
+Now that the build logic has been translated into a plugin, you can apply it in the build script.
+The build script has been shrunk from 8 lines of code to a one liner.
 
 ++++
 <sample id="conditionalLogicDo" dir="userguide/bestPractices/conditionalLogic/do" title="A build script applying a plugin that encapsulates imperative logic">
     <sourcefile file="build.gradle"/>
 </sample>
 ++++
 
+[[sec:avoiding_use_of_gradlebuild]]
+=== Avoiding the use of `GradleBuild`
+
+The api:org.gradle.api.tasks.GradleBuild[] task type allows a build script to define a task that invokes another Gradle build.
+The use of this type is generally discouraged.
+There are some corner cases where the invoked build doesn't expose the same runtime behavior as from the command line or through the Tooling API leading to unexpected results.
+
+Usually, there's a better way to model the requirement.
+The appropriate approach depends on the problem at hand. Here're some options:
+
+* Model the build as <<multi_project_builds,multi-project build>> if the intention is to execute tasks from different modules as unified build.
+* Use <<composite_builds,composite builds>> for projects that are physically separated but should occasionally be built as a single unit.
+
+[[sec:avoiding_inter_project_configuration]]
+=== Avoiding inter-project configuration
+
+Gradle does not restrict build script authors from reaching into the domain model from one project into another one in a <<multi_project_builds,multi-project build>>.
+Strongly-coupled projects hurts <<sec:parallel_execution,build execution performance>> as well as readability and maintainability of code.
+
+The following practices should be avoided:
```

Comment:
We should provide guidance on what users should do instead for each of these points, where possible.

Reply:
I absolutely agree. In this case it's not straightforward to explain the correct alternative with a sentence or two. Readers will not get what we mean by it.

It will require explaining the use case, a proper code example + how to solve the problem with the right approach and _why_. IMHO we'd need to revamp the multi-project chapter completely to achieve that. It would also involve deleting existing content that actually explains the anti-pattern _right now_.

Reply:
Hmm, can we say the solution depends on the context and suggest some options just as we do in the "Avoiding the use of `GradleBuild`", or is that not really doable?

Reply:
Would you personally understand when I say something along the lines of "You will need to create a custom configuration, create an archive with the compiled byte code and declare it as outgoing artifact for the project by using the configuration. In the consuming project you will have to declare a dependency on the outgoing artifact by referencing the configuration."

I am 100% sure readers won't understand this without an example. And that would probably only cover one of the use cases.

Reply:
I generally get what that means, but if you asked me to translate that into code without other docs I could not. 

I guess since you seem to feel strongly that this isn't worth the massive effort, we can leave as-is.

Reply:
Don't get me wrong - we should definitely do it. I just think it makes more sense as a different PR that also aligns other references to it at the same time. And given that the effort isn't just half and hour to an hour I'd rather do it separately.